src/pllapallpal/Logic.java

```diff
@@ -16,14 +15,21 @@ public Logic() {
 
     public void readFile() {
```

Comment:
Try this.
<pre>
try (BufferedReader reader = new BufferedReader(new FileReader(PATH))) {

    reader.lines().forEach(str -> {
        // do Something.
    });

} catch (IOException e) {
    e.printStackTrace();
}
</pre>